src/rt/lifetime.d

```diff
@@ -1116,7 +1086,44 @@ extern (C) void[] _d_newarraymiT(const TypeInfo ti, size_t ndims, ...)
             va_start(q, __va_argsave);
         else
             static assert(false, "platform not supported");
-        return _d_newarrayOpT!(_d_newarrayiT)(ti, ndims, q);
+
+        auto dims = (cast(size_t*)alloca(size_t.sizeof * ndims))[0..ndims];
+        foreach(ref v; dims)
+            va_arg(q, v);
+        va_end(q);
+        return _d_newarrayOpT!(_d_newarrayiT)(ti, dims);
+    }
+}
+
+
+/**
+ *
+ */
+extern (C) void[] _d_newarraymTX(const TypeInfo ti, size_t[] dims)
+{
+    debug(PRINTF) printf("_d_newarraymT(dims.length = %d)\n", dims.length);
+
+    if (dims.length == 0)
+        return null;
+    else
+    {
+        return _d_newarrayOpT!(_d_newarrayT)(ti, dims);
+    }
+}
+
+
+/**
+ *
+ */
+extern (C) void[] _d_newarraymiTX(const TypeInfo ti, size_t[] dims)
```

Comment:
Oh ok, that makes more sense.
The whole AA mess has made me wary of higher-level lowerings, although it's the only way to get dmd's inliner to look at runtime calls.  I think in the long run we'll drop the typeinfo interface, but I'm not in a hurry.